subsys/usb/class/netusb/netusb.c

```diff
@@ -196,21 +196,31 @@ static void netusb_init(struct net_if *iface)
 	}
 
 #ifndef CONFIG_USB_COMPOSITE_DEVICE
-	int ret;
+	/* Linker-defined symbols bound the USB descriptor structs */
+	extern struct usb_cfg_data __usb_data_start[];
+	extern struct usb_cfg_data __usb_data_end[];
+	size_t size = (__usb_data_end - __usb_data_start);
 
-	netusb_config.interface.payload_data = interface_data;
-	netusb_config.usb_device_description = usb_get_device_descriptor();
+	for (size_t i = 0; i < size; i++) {
```

Comment:
We should revise and merge the "legacy" and composite initialization, see als #10503